proxy/proxy.go

```diff
@@ -83,17 +87,84 @@ func (proxy *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	proxy.Intercept(i, w, r)
 }
 
-func (proxy *Proxy) ListenAndServe() error {
-	listener, err := net.Listen("tcp", proxy.ListenAddr)
+func (proxy *Proxy) ListenAndServe() {
+	errs := make(chan error)
+	for _, addr := range proxy.ListenAddrs {
+		Info.Println("proxy listening on", addr)
+	}
```

Comment:
This breaks the intended semantics of "wait_for_log` in the weave script.

You really ought to get all the listening (and associated error checking) out of the way *before* printing out any "proxy listening on".